File: js/src/CoreAbstraction/IBlockchainActivities.ts
import { BalanceRequest } from "./Models/BalanceRequestModels/BalanceRequest";
import { BalanceResponse } from "./Models/BalanceRequestModels/BalanceResponse";
import { BaseRequest } from "./Models/BaseRequest";
import { BlockNumberResponse } from "./Models/BlockNumberResponse";
import { HTLCBlockEventResponse } from "./Models/EventModels/HTLCBlockEventResposne";
import { EventRequest } from "./Models/EventRequest";
import { AddLockSignatureRequest } from "./Models/TransactionBuilderModels/AddLockSignatureRequest";

export interface IBlockchainActivities {
    GetBalanceAsync(request: BalanceRequest): Promise<BalanceResponse>;  
  
    GetLastConfirmedBlockNumberAsync(request: BaseRequest): Promise<BlockNumberResponse>;  
    
    ValidateAddLockSignatureAsync(request: AddLockSignatureRequest): Promise<boolean>;
    
    GetEventsAsync(request: EventRequest): Promise<HTLCBlockEventResponse>;
  }

File: csharp/src/Workflows.Abstractions/Activities/IBlockchainActivities.cs
using Temporalio.Activities;
using Train.Solver.Workflows.Abstractions.Models;

namespace Train.Solver.Workflows.Abstractions.Activities;

public interface IBlockchainActivities
{
    [Activity]
    Task<BalanceResponse> GetBalanceAsync(BalanceRequest request);

    [Activity]
    Task<BlockNumberResponse> GetLastConfirmedBlockNumberAsync(BaseRequest request);

    [Activity]
    Task<bool> ValidateAddLockSignatureAsync(AddLockSignatureRequest request);

    [Activity]
    Task<HTLCBlockEventResponse> GetEventsAsync(EventRequest request);
}
